include/solvers/petsc_nonlinear_solver.h

```diff
@@ -158,6 +179,22 @@ class PetscNonlinearSolver : public NonlinearSolver<T>
    */
   void use_default_monitor(bool state) { _default_monitor = state; }
 
+  /**
+   * Abstract base class to be used to implement a custom line-search algorithm
+   */
+  class ComputeLineSearchObject
+  {
+  public:
+    virtual ~ComputeLineSearchObject () {}
+
+    virtual void linesearch (SNESLineSearch linesearch) = 0;
+  };
+
+  /**
+   * A callable object that can be used to specify a custom line-search
+   */
+  std::unique_ptr<ComputeLineSearchObject> linesearch_object;
```

Comment:
Back when we were using "fake" C++03 unique_ptr stand-ins, it was bad to put them in the public interface like this because it was possible for them to be accidentally "stolen" too easily. For example, the `BoundaryInfo` in `MeshBase` is only accessible through the `get_boundary_info()` interface. Now that we are using real unique_ptrs it's probably fine though?

@roystgnr any opinion on this?

Reply:
I think we decided to just go with this change, so this comment should be considered "resolved"